packages/datadog-plugin-winston/src/index.js

```diff
@@ -15,7 +15,8 @@ function chunkProxy (chunk, holder) {
       }
     },
     ownKeys (target) {
-      return ['dd', ...Reflect.ownKeys(target)]
+      const ownKeys = Reflect.ownKeys(target)
```

Comment:
Wouldn't the `in` operator be sufficient here? It would avoid the array conversion and the slower `includes`.

Reply:
No. We always need at least one call to `Reflect.ownKeys()` in order to correctly pass the ownKeys through the trap. The conversion to array always happens regardless.

`in` is insufficient because it includes the prototype chain, which is _not_ how ownKeys works:

```
> 'foo' in ['foo', 'bar']
false
> a = { bar: 1, __proto__: { foo: 2 } }
{ bar: 1 }
> 'foo' in a
true
> Reflect.ownKeys(a)
[ 'bar' ]
```

Reply:
It's fine to include the prototype chain. If `dd` is available _at all_ we should not overwrite it.

Reply:
> It's fine to include the prototype chain.

This is the `ownKeys` proxy trap, where it's _not_ fine to include the prototype chain.

>  If dd is available at all we should not overwrite it, so I would actually argue that ownKeys is insufficient.

Assuming that by "overwrite" you mean "overwrite via proxy, not for real", then the answer ought to be different for the prototype chain, where those properties would never be stringified in the first place.